### PR TITLE
java-security offers adapter to read from spring-xsuaa SecurityContext

### PR DESCRIPTION
**Pull request by nenaraab**

### CODE REVIEW

File: java-security/src/main/java/com/sap/cloud/security/adapter/spring/SpringSecurityContext.java [hunk 1]
```diff
@@ -21,36 +24,86 @@
...
+ * Use this class in case your application sets Spring's security context via
+ * one of these libraries: <br>
+ * <ol>
+ * <li>@code{org.springframework.security.oauth:spring-security-oauth2} or</li>
+ * <li>@code{com.sap.cloud.security.xsuaa:spring-xsuaa} client library.</li>
+ * </ol>
  */
 public class SpringSecurityContext {
 
+	static Logger LOGGER = LoggerFactory.getLogger(SpringSecurityContext.class);
```
Comment (liga-oz) [inline java-security/src/main/java/com/sap/cloud/security/adapter/spring/SpringSecurityContext.java:37]: as it's not final field variable name should be in lower case I think.

File: java-security/src/main/java/com/sap/cloud/security/adapter/spring/SpringSecurityContext.java [hunk 2]
```diff
@@ -21,36 +24,86 @@
...
-			OAuth2AuthenticationDetails authDetails = (OAuth2AuthenticationDetails) authentication.getDetails();
-			String tokenValue = authDetails.getTokenValue();
-			AbstractToken xsuaaToken = new XsuaaTokenWithGrantedAuthorities(tokenValue,
-					authentication.getAuthorities());
-			if (xsuaaToken.isXsuaaToken()) {
-				return xsuaaToken;
+		if (Objects.nonNull(authentication) && authentication.isAuthenticated()) {
+			try {
+				if (authentication.getDetails() != null && authentication.getDetails().getClass()
+						.getName() == "org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationDetails") {
```
Comment (liga-oz) [inline java-security/src/main/java/com/sap/cloud/security/adapter/spring/SpringSecurityContext.java:54]: I would suggest using equals() method from String class for comparison.

File: java-security/src/test/java/com/sap/cloud/security/adapter/spring/SpringXsuaaSecurityContextTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+		assertThat(tokenInOtherThread.get()).isNull();
+	}
+
+	@Test
+	public void clearingTokenInDifferentThreadDoesNotAffectMainThread()
+			throws ExecutionException, InterruptedException {
+		setToken(token);
+
+		executorService.submit(() -> SpringSecurityContext.clear()).get(); // run and await other thread
```
Comment (liga-oz) [inline java-security/src/test/java/com/sap/cloud/security/adapter/spring/SpringXsuaaSecurityContextTest.java:96]: you can shorten the lambda expression -> executorService.submit(SpringSecurityContext::clear).get()

**Review by liga-oz — APPROVED** [review]
just few minor suggestions

File: java-security/src/test/java/com/sap/cloud/security/adapter/spring/SpringXsuaaSecurityContextTest.java (hunk 3 above)
Comment (nenaraab) [reply java-security/src/test/java/com/sap/cloud/security/adapter/spring/SpringXsuaaSecurityContextTest.java:96]: cool 🥇

**Pull request merged**